pkg/apis/operator/v1alpha1/tektonaddon_validation.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2021 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+
+	"knative.dev/pkg/apis"
+)
+
+func (ta *TektonAddon) Validate(ctx context.Context) (errs *apis.FieldError) {
+
+	if apis.IsInDelete(ctx) {
+		return nil
+	}
+
+	if ta.Spec.TargetNamespace == "" {
```

Comment:
shall we default (in the defaulter) to the `DEFAULT_TARGET_NAMESPACE=openshift-pipelines` from config-defaults. If we don't do it, what is the purpose of `DEFAULT_TARGET_NAMESPACE=openshift-pipelines`

Reply:
oh.. `DEFAULT_TARGET_NAMESPACE=openshift-pipelines` for autocreate TektonConfig. 
So i think returning error is the safest thing here 👍 .